docs/proposed/http-storage-node-protocol.rst

```diff
@@ -701,7 +700,10 @@ Immutable Data
 Mutable Data
 ~~~~~~~~~~~~
 
-1. Create mutable share number ``3`` with ``10`` bytes of data in slot ``BBBBBBBBBBBBBBBB``::
+1. Create mutable share number ``3`` with ``10`` bytes of data in slot ``BBBBBBBBBBBBBBBB``.
+The special test vector of size 1 but empty bytes will only pass
```

Comment:
I think this suggests we should clarify the behavior for the case of an over-read attempt.  If size is greater than data available to be read from offset, is it an error in general and only the special case of offset=0 size=1 compared against b"" is allowed?  Or is it the case that an over-read attempt results in a short read and whatever data was found is what is compared?

I don't know what the use-case is for anything except detecting non-existing shares but I think the latter still makes more sense: having a general rule leads to fewer surprises than having a single special case.

Reply:
Added this to the reference section:

> A test vector or read vector that read beyond the boundaries of existing data will return nothing for any bytes past the end.
> As a result, if there is no data at all, an empty bytestring is returned no matter what the offset or length.

Reply:
I.e. made short-reads-return-empty-bytestring the documented behavior (which matches Foolscap code.)